packages/framework/src/Model/Product/Search/Export/ProductSearchExportStructureFacade.php

```diff
@@ -29,20 +38,80 @@ public function __construct(Domain $domain, ElasticsearchStructureManager $elast
         $this->elasticsearchStructureManager = $elasticsearchStructureManager;
     }
 
+    /**
+     * @required
+     * @param \Shopsys\FrameworkBundle\Component\Elasticsearch\ElasticsearchStructureUpdateChecker $elasticsearchStructureUpdateChecker
+     * @internal Will be replaced with constructor injection in the next major release
+     */
+    public function setElasticsearchStructureUpdateChecker(ElasticsearchStructureUpdateChecker $elasticsearchStructureUpdateChecker): void
+    {
+        if ($this->elasticsearchStructureUpdateChecker !== null) {
+            throw new BadMethodCallException(sprintf('Method "%s" has been already called and cannot be called multiple times.', __METHOD__));
+        }
+
+        $this->elasticsearchStructureUpdateChecker = $elasticsearchStructureUpdateChecker;
+    }
+
+    /**
+     * @param \Symfony\Component\Console\Output\OutputInterface $output
+     */
+    public function migrateIndexesIfNecessary(OutputInterface $output)
+    {
+        foreach ($this->domain->getAllIds() as $domainId) {
+            $output->writeln(sprintf('Migrating index for domain with ID %s', $domainId));
+
+            $index = ProductElasticsearchRepository::ELASTICSEARCH_INDEX;
+
+            if ($this->elasticsearchStructureUpdateChecker->isNecessaryToUpdateStructure($domainId, $index)) {
+                $this->elasticsearchStructureManager->migrate($domainId, $index);
+                $output->writeln(sprintf('Migrating done for domain with ID %s', $domainId));
+            } else {
+                $output->writeln(sprintf('Migrating is not necessary as there were no changes in the definition'));
+            }
+        }
+    }
+
     /**
      * @param \Symfony\Component\Console\Output\OutputInterface $output
      */
     public function createIndexes(OutputInterface $output)
     {
         foreach ($this->domain->getAllIds() as $domainId) {
-            $output->writeln(sprintf('Creating index for id %s', $domainId));
+            $this->createIndexIfNecessary($output, $domainId);
+            $this->createAlias($output, $domainId);
+        }
+    }
 
-            $this->elasticsearchStructureManager->createIndex(
-                $domainId,
-                ProductElasticsearchRepository::ELASTICSEARCH_INDEX
-            );
+    /**
+     * @param \Symfony\Component\Console\Output\OutputInterface $output
+     * @param int $domainId
+     */
+    protected function createAlias(OutputInterface $output, int $domainId)
+    {
+        $output->writeln(sprintf('Creating alias for domain with ID %s', $domainId));
+
+        $this->elasticsearchStructureManager->createAliasForIndex(
+            $domainId,
+            ProductElasticsearchRepository::ELASTICSEARCH_INDEX
+        );
+
+        $output->writeln('Alias created');
```

Comment:
I'm not sure with the message here as it seems that the alias is created only if it doesn't already exist...

Reply:
I removed it